tensorflow/compiler/tf2tensorrt/convert/convert_nodes.cc

```diff
@@ -2597,6 +2687,203 @@ Status ConvertConv2DBackpropInput(OpConverterParams* params) {
   return ConvertConv2DHelper(params, 1, /*is_conv2d_backprop_input=*/true);
 }
 
+#if IS_TRT_VERSION_GE(6, 0, 0, 0)
+Status ConvertConv3DHelper(OpConverterParams* params, int group,
+                           bool is_conv3d_backprop_input = false) {
+  const int kNUM_DIMS = 5;
+  const auto& inputs = params->inputs;
+  const auto& node_def = params->node_def;
+  TRT_TensorOrWeights backprop_output_size;
+  nvinfer1::ITensor* tensor = nullptr;
+  if (is_conv3d_backprop_input) {
+    // In the case when Conv3dBackpropInput is used for conv3d_transpose, these
+    // inputs correspond to: output size, filter, and input.
+    TF_RETURN_IF_ERROR(CheckInputsWeights(
+        *params,
+        {{"input_sizes", true}, {"filter", true}, {"out_backprop", false}}));
+    backprop_output_size = inputs.at(0);
+    tensor = inputs.at(2).tensor();
+  } else {
+    TF_RETURN_IF_ERROR(
+        CheckInputsWeights(*params, {{"input", false}, {"filter", true}}));
+    tensor = inputs.at(0).tensor();
+  }
+  TF_RETURN_IF_ERROR(
+      AllowDataTypes(*params, {DataType::DT_FLOAT, DataType::DT_HALF}));
+  const TRT_ShapedWeights weights_drsck = inputs.at(1).weights();
+  if (weights_drsck.shape_.nbDims != kNUM_DIMS) {
+    return errors::InvalidArgument("Conv3D expects kernel of dimension 5, at " +
+                                   node_def.name());
+  }
+  TFAttrs attrs(node_def);
+  auto data_format = attrs.get<string>("data_format");
+  const bool isNDHWC = (data_format == "NDHWC");  // Or NCDHW 01234 - > 02341
+  const int d_index = isNDHWC ? 1 : 2;
+  const int h_index = isNDHWC ? 2 : 3;
+  const int w_index = isNDHWC ? 3 : 4;
+  const int c_index = isNDHWC ? 4 : 1;
+  auto tf_dilations = attrs.get<std::vector<int64>>("dilations");
+  if (tf_dilations.size() != kNUM_DIMS) {
+    return errors::InvalidArgument(
+        "Convolution dilations field must specify 5 dimensions, at ",
+        node_def.name());
+  }
+  if (tf_dilations[0] != 1 || tf_dilations[c_index] != 1) {
+    return errors::Unimplemented(
+        "Dilation rate must be 1 for batch and channel dimensions, at ",
+        node_def.name());
+  }
+
+  const nvinfer1::Dims3 dilationDHW(
```

Comment:
Please fix the variable names. All variables that are not compile-time constants should use snakecase.